hot/style-loader.js

```diff
@@ -20,7 +20,18 @@ module.exports.pitch = function (request) {
       const updatePanelElems = require('panel/hot/update-panel-elems');
       updatePanelElems('${elemName}', elem => {
         if (elem.getConfig('useShadowDom')) {
-          elem.el.querySelector('style').textContent = newStyle.toString();
+          const newStyleText = newStyle.toString();
+          if (elem.el.adoptedStyleSheets) {
+            if (newStyleText) {
+              const newStylesheet = new CSSStyleSheet();
+              newStylesheet.replaceSync(newStyleText);
+              elem.el.adoptedStyleSheets = [newStylesheet, ...elem.el.adoptedStyleSheets.slice(1)];
```

Comment:
why appending rather than changing the stylesheet content here?

Reply:
to simplify it so that it implicitly includes the case where the old style was empty (which means there won't be a stylesheet object to modify). There isn't always a stylesheet object there.

Reply:
hm the logic still isn't quite right in the case where the old css config is empty but `style-overrides` is not empty....gotta fix that

Reply:
just updated the replace sync, sorry for the last minute tweak

Reply:
hmm, one more thing: the cache is not updated. so unmount / mount after hot reload might causing revert of the old style

Reply:
You're right, the `style-override` seemed to be still a bit broken, I've just updated the PR so it should be fixed now. I've also made sure to update the stylesheet cache.